scripting/togglemusic_dhook.sp

```diff
@@ -33,81 +27,89 @@ StringMap g_smCommon;
+	
 	RegConsoleCmd("sm_music", Command_Music, "Toggles map music");
 	RegConsoleCmd("sm_stopmusic", Command_StopMusic, "Toggles map music");
 	RegConsoleCmd("sm_volume", Command_Volume, "Brings volume menu");
-
+	
 	g_ConVar_Debug = CreateConVar("sm_togglemusic_debug", "0", "Debug mode (0 = off, 1 = on)", 0, true, 0.0, true, 1.0);
 	g_iDebug = GetConVarInt(g_ConVar_Debug);
 	HookConVarChange(g_ConVar_Debug, OnConVarChanged);
-
+	
 	if (g_smSourceEnts == null)
 		g_smSourceEnts = new StringMap();
-	
+		
 	if (g_smChannel == null)
 		g_smChannel = new StringMap();
-	
+		
 	if (g_smCommon == null)
 		g_smCommon = new StringMap();
-	
+		
 	if (g_smRecent == null)
 		g_smRecent = new StringMap();
 		
 	if (g_smVolume == null)
 		g_smVolume = new StringMap();
-	
+		
 	if (g_hClientVolCookie == null)
 		g_hClientVolCookie = RegClientCookie("togglemusic_volume", "ToggleMusic Volume Pref", CookieAccess_Protected);
 		
 	if (g_hClientMusicCookie == null)
 		g_hClientMusicCookie = RegClientCookie("togglemusic_music", "ToggleMusic Music Pref", CookieAccess_Protected);
-	
+		
 	char preError[256];
 	char prePattern[256] = "(([-_a-zA-Z0-9]+[/]?)+[.][a-zA-Z0-9]{3})";
 	regPattern = CompileRegex(prePattern, PCRE_CASELESS, preError, sizeof(preError), regError);
-	if (regError != REGEX_ERROR_NONE) { LogError(preError); }
-	
+	if (regError != REGEX_ERROR_NONE)
+		LogError(preError);
+		
 	if (hAcceptInput == null)
 	{
-	
+		
 		EngineVersion eVer = GetEngineVersion();
 		char tmpOffset[148];
 		
-		if (eVer == Engine_CSGO) {
+		if (eVer == Engine_CSGO)
 			tmpOffset = "sdktools.games\\engine.csgo";
-		} else if (eVer == Engine_CSS) {
-				tmpOffset = "sdktools.games\\engine.css";
-		} else if (eVer == Engine_TF2) {
-				tmpOffset = "sdktools.games\\engine.tf";
-		} else if (eVer == Engine_Contagion) {
-				tmpOffset = "sdktools.games\\engine.contagion";
-		} else if (eVer == Engine_Left4Dead2) {
-				tmpOffset = "sdktools.games\\engine.Left4Dead2";
-		} else if (eVer == Engine_AlienSwarm) {
-				tmpOffset = "sdktools.games\\engine.swarm";
-		}
-		
+			
+		else if (eVer == Engine_CSS)
+			tmpOffset = "sdktools.games\\engine.css";
+			
+		else if (eVer == Engine_TF2)
+			tmpOffset = "sdktools.games\\engine.tf";
+			
+		else if (eVer == Engine_Contagion)
+			tmpOffset = "sdktools.games\\engine.contagion";
+			
+		else if (eVer == Engine_Left4Dead2)
+			tmpOffset = "sdktools.games\\engine.Left4Dead2";
+			
+		else if (eVer == Engine_AlienSwarm)
+			tmpOffset = "sdktools.games\\engine.swarm";
+			
 		Handle temp = LoadGameConfigFile(tmpOffset);
-	
-		if(temp == null) {
-			SetFailState("Why you no has gamedata?");
-		}
-
-		HookEvent("round_poststart", Event_PostRoundStart);
 		
+		if (temp == null)
+			SetFailState("Gamedata Missing!");
+			
+		if (eVer == Engine_CSGO)
+			HookEvent("round_poststart", Event_RoundStarted); // CSGO Only
+			
+		else
+			HookEvent("round_start", Event_RoundStarted); // Supported on all Games
```

Comment:
Isn't round_start a bit too early in many instances, hence the need for round_poststart? The intention is good, however this should require testing to actually confirm this change is needed, or even works.

On the other hand, this could also be used and specific games might not need to be defined, as round_start might be good enough for all. Generally speaking, if round_start works for one game, it should work for most/all.

Reply:
If its too early we could use RequestFrame2, which is RequestFrame just with the mount of Frames to specify. The Forum post Title starts with "[ANY]" meaning that it works with most HL Mods aswell as all updated Valve Games. CSS and TF2 should be fine for testing, if it makes Problems with other Mods, a Ticket will be opened.